runtime/compiler/ilgen/ClassLookahead.cpp

```diff
@@ -1368,18 +1368,17 @@ void TR_ClassLookahead::makeInfoPersistent()
             {
             if (_traceIt)
                printf("Creating persistent info for array field %s\n", persistentSig);
-            newFieldInfo = new (PERSISTENT_NEW) TR_PersistentArrayFieldInfo(persistentSig, length);
-            memcpy(newFieldInfo, fieldInfo, sizeof(TR_PersistentArrayFieldInfo));
+            newFieldInfo = new (PERSISTENT_NEW) TR_PersistentArrayFieldInfo(*arrayFieldInfo);
```

Comment:
Using `memcpy` to clone a C++ object is undefined behaviour and may overwrite the VFT!